pkg/proxy/handlers/spacelister.go

```diff
@@ -61,57 +96,16 @@ func (s *SpaceLister) HandleSpaceListRequest(ctx echo.Context) error {
 		// specific workspace requested so add label requirement to match the space
 		spaceSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingSpaceLabelKey, selection.Equals, []string{workspaceName})
 		if err != nil {
-			log.Error(nil, err, "error constructing space selector")
-			return errorResponse(ctx, apierrors.NewInternalError(fmt.Errorf("workspace lookup failed")), http.StatusInternalServerError)
+			return nil, err
 		}
 		requirements = append(requirements, *spaceSelector)
 	}
 
 	spaceBindings, err := s.GetInformerServiceFunc().ListSpaceBindings(requirements...)
 	if err != nil {
-		log.Error(nil, err, "error listing space bindings")
-		return errorResponse(ctx, apierrors.NewInternalError(fmt.Errorf("workspace lookup failed")), http.StatusInternalServerError)
-	}
-
-	workspaces := s.workspacesFromSpaceBindings(signup.Name, spaceBindings)
-
-	if doGetWorkspace { // specific workspace requested
-		if len(workspaces) != 1 {
-			// not found
-			r := schema.GroupResource{Group: "toolchain.dev.openshift.com", Resource: "workspaces"}
-			return errorResponse(ctx, apierrors.NewNotFound(r, workspaceName), http.StatusNotFound)
-		}
-		return getWorkspaceResponse(ctx, workspaces[0])
-	}
-
-	return listWorkspaceResponse(ctx, workspaces)
-}
-
-func errorResponse(ctx echo.Context, err *apierrors.StatusError, code int) error {
-	ctx.Logger().Error(err, "workspace list error")
-	ctx.Response().Writer.Header().Set("Content-Type", "application/json")
-	ctx.Response().Writer.WriteHeader(code)
-	return json.NewEncoder(ctx.Response().Writer).Encode(err.ErrStatus)
-}
-
-func getWorkspaceResponse(ctx echo.Context, workspace toolchainv1alpha1.Workspace) error {
-	ctx.Response().Writer.Header().Set("Content-Type", "application/json")
-	ctx.Response().Writer.WriteHeader(http.StatusOK)
-	return json.NewEncoder(ctx.Response().Writer).Encode(workspace)
-}
-
-func listWorkspaceResponse(ctx echo.Context, workspaces []toolchainv1alpha1.Workspace) error {
-	workspaceList := &toolchainv1alpha1.WorkspaceList{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "WorkspaceList",
-			APIVersion: "toolchain.dev.openshift.com/v1alpha1",
-		},
-		Items: workspaces,
+		return nil, err
 	}
-
-	ctx.Response().Writer.Header().Set("Content-Type", "application/json")
-	ctx.Response().Writer.WriteHeader(http.StatusOK)
-	return json.NewEncoder(ctx.Response().Writer).Encode(workspaceList)
+	return spaceBindings, nil
```

Comment:
minor detail, another option could be:

```
return s.GetInformerServiceFunc().ListSpaceBindings(requirements...)
```

Reply:
True! I think I do that way too often haha. Done in f675b9238999e4d587e5cf17cb28f52c890cf8ce